AppInfo.swift

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+public class AppInfo {
+    /// Return the shared container identifier (also known as the app group) to be used with for example background
+    /// http requests. It is the base bundle identifier with a "group." prefix.
+    public static func sharedContainerIdentifier() -> String? {
+        if let baseBundleIdentifier = AppInfo.baseBundleIdentifier() {
+            return "group." + baseBundleIdentifier
+        }
+        return nil
+    }
+
+    /// Return the base bundle identifier.
+    ///
+    /// This function is smart enough to find out if it is being called from an extension or the main application. In
+    /// case of the former, it will chop off the extension identifier from the bundle since that is a suffix not part
+    /// of the *base* bundle identifier.
+    public static func baseBundleIdentifier() -> String? {
+        let bundle = NSBundle.mainBundle()
+        if let packageType = bundle.objectForInfoDictionaryKey("CFBundlePackageType") as? NSString {
+            if let baseBundleIdentifier = bundle.bundleIdentifier {
+                if packageType == "XPC!" {
+                    let components = baseBundleIdentifier.componentsSeparatedByString(".")
+                    return components[0..<components.count-1].joinWithSeparator(".")
+                }
+                return baseBundleIdentifier
+            }
+        }
+        return nil
```

Comment:
I'd change this assume that `packageType` is always non-nil, then change the function to return a `String` instead of a `String?`. Things will break in strange ways if we can't get the bundle ID (settings won't work, the blockers won't load, etc.). That's no better than simply crashing.

As a bonus, if you remove the optional return value here, `sharedContainerIdentifier`, `Settings.getBool`, etc., there will be a little less code since we don't have to have checks for failure cases.